sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -601,6 +600,57 @@ object ExecChecks {
   def hiddenHack() = new ExecChecks(TypeSig.all, TypeSig.all, shown = false)
 }
 
+/**
+ * Base class all Partition checks must follow
+ */
+abstract class PartChecks extends TypeChecks[Map[String, SupportLevel]]
+
+case class PartChecksImpl(
+    paramCheck: Seq[ParamCheck] = Seq.empty,
+    repeatingParamCheck: Option[RepeatingParamCheck] = None)
+    extends PartChecks {
+
+  override def tag(meta: RapidsMeta[_, _, _]): Unit = {
+    val part = meta.wrapped
+    val children = meta.childExprs.map(_.wrapped.asInstanceOf[Expression]).toArray
+
+    val fixedChecks = paramCheck.toArray
+    assert (fixedChecks.length <= children.length,
+      s"${part.getClass.getSimpleName} expected at least ${fixedChecks.length} but " +
+          s"found ${children.length}")
+    fixedChecks.indices.foreach { i =>
+      val check = fixedChecks(i)
```

Comment:
Nit: can simplify with something like this:
```suggestion
    fixedChecks.zipWithIndex.foreach { case (check, i) =>
```